### PR TITLE
LAU-1183 sanitize string before performing search

### PR DESCRIPTION
**Pull request by priyankaVerma21**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/LAU-1183

### Change description ###
Sanitise String before performing search

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/controllers/BaseSearchController.ts [hunk 1]
```diff
@@ -112,9 +112,21 @@ export abstract class BaseSearchController<SearchType extends SearchTypeCommon>
...
 
   formatSearchRequest(request: Partial<SearchType>): void {
-    // Remove any properties with empty strings from the request object
-    // @ts-ignore
-    Object.keys(request).forEach(key => request[key] === '' ? delete request[key] : {});
+  // Remove any properties with empty strings from the request object
+    Object.keys(request).forEach(key => {
+      const typedKey = key as keyof SearchType;
+      if (request[typedKey] === '') {
+        delete request[typedKey];
```
Comment (MartinYSpasov) [inline src/main/controllers/BaseSearchController.ts:119]: Do we pass empty strings as part of the search

**Review by MartinYSpasov — CHANGES_REQUESTED** [review]

File: src/main/controllers/BaseSearchController.ts (hunk 1 above)
Comment (priyankaVerma21) [reply src/main/controllers/BaseSearchController.ts:119]: Also If I remove this code all the unit test start failing as at the moment it's only passing the parameters where we don't have empty string , for example all the test case which added as part of this and any previous one

**Review by MartinYSpasov — APPROVED** [review]
